include/utils.h

```diff
@@ -23,6 +23,7 @@ typedef struct {
 typedef struct {
     char *value;
     int error_code;
+    int malloced;
```

Comment:
Why did you introduce this variable. Is there any downside to use malloc for all strings?
The variable name is a little misleading, when `malloced` is true(>0) I would assume the value string to be allocated by malloc. But the variable is used the other way around.